python/llm/src/bigdl/llm/transformers/speculative.py

```diff
@@ -271,29 +276,59 @@ def speculative_generate(self,
         else:
             draft_current_input_ids = current_input_ids
             # Target model KV cache to draft model
-            draft_past_key_values = past_key_values
+            if _enable_ipex:
+                cur_len = past_key_values[0][0].size(1)
+                tmp_draft_past_key_values = [
+                    [pkv[1].permute(1, 2, 0, 3)[:,:,:cur_len,:], pkv[2].permute(1, 2, 0, 3)[:,:,:cur_len,:]]
+                    for pkv in past_key_values
+                ]
+
+                def _expand(tmp_draft_past_key_values):
+                    s0, s1, s2, s3 = tmp_draft_past_key_values.size()
+                    draft_past_key_values_storage = torch.zeros(s0, s1, s2 + 256, s3, dtype=tmp_draft_past_key_values.dtype, device=tmp_draft_past_key_values.device)
+                    draft_past_key_values_storage[:, :, :s2, :] = tmp_draft_past_key_values
+                    draft_past_key_values = draft_past_key_values_storage[:, :, :s2, :]
+                    return draft_past_key_values
+
+                tmp_draft_past_key_values = [
+                    (k.to(torch.float32), v.to(torch.float32)) for k, v in tmp_draft_past_key_values
+                ]
+                draft_past_key_values = [
+                    [_expand(pkv[0]), _expand(pkv[1])] for pkv in tmp_draft_past_key_values
```

Comment:
Each draft_step == 0 always call _expand method will causing preformance decrease. And need to solve conflict with [this pr](https://github.com/intel-analytics/BigDL/pull/9964)